diff --git a/packages/frontend/core/src/desktop/dialogs/setting/general-setting/editor/edgeless/connector.tsx b/packages/frontend/core/src/desktop/dialogs/setting/general-setting/editor/edgeless/connector.tsx
index 650cc92977..dbedee7bed 100644
--- a/packages/frontend/core/src/desktop/dialogs/setting/general-setting/editor/edgeless/connector.tsx
+++ b/packages/frontend/core/src/desktop/dialogs/setting/general-setting/editor/edgeless/connector.tsx
@@ -127,14 +127,27 @@ export const ConnectorSettings = () => {
     () => [
       {
         value: StrokeStyle.Solid,
+        testId: 'connector-border-style-solid-trigger',
         label:
           t['com.affine.settings.editorSettings.edgeless.note.border.solid'](),
       },
       {
         value: StrokeStyle.Dash,
+        testId: 'connector-border-style-dash-trigger',
         label:
           t['com.affine.settings.editorSettings.edgeless.note.border.dash'](),
       },
+      {
+        value: StrokeStyle.Dot,
+        testId: 'connector-border-style-dot-trigger',
+        label: 'Dot',
+      },
+      {
+        value: StrokeStyle.None,
+        testId: 'connector-border-style-none-trigger',
+        label:
+          t['com.affine.settings.editorSettings.edgeless.note.border.none'](),
+      },
     ],
     [t]
   );
@@ -443,6 +456,7 @@ export const ConnectorSettings = () => {
           max={12}
           step={2}
           nodes={[2, 4, 6, 8, 10, 12]}
+          disabled={borderStyle === StrokeStyle.None}
         />
       </SettingRow>
       <SettingRow
diff --git a/packages/frontend/core/src/desktop/dialogs/setting/general-setting/editor/edgeless/note.tsx b/packages/frontend/core/src/desktop/dialogs/setting/general-setting/editor/edgeless/note.tsx
index b5b6ca278c..c9cc455bcd 100644
--- a/packages/frontend/core/src/desktop/dialogs/setting/general-setting/editor/edgeless/note.tsx
+++ b/packages/frontend/core/src/desktop/dialogs/setting/general-setting/editor/edgeless/note.tsx
@@ -58,16 +58,24 @@ export const NoteSettings = () => {
     () => [
       {
         value: StrokeStyle.Solid,
+        testId: 'note-border-style-solid-trigger',
         label:
           t['com.affine.settings.editorSettings.edgeless.note.border.solid'](),
       },
       {
         value: StrokeStyle.Dash,
+        testId: 'note-border-style-dash-trigger',
         label:
           t['com.affine.settings.editorSettings.edgeless.note.border.dash'](),
       },
+      {
+        value: StrokeStyle.Dot,
+        testId: 'note-border-style-dot-trigger',
+        label: 'Dot',
+      },
       {
         value: StrokeStyle.None,
+        testId: 'note-border-style-none-trigger',
         label:
           t['com.affine.settings.editorSettings.edgeless.note.border.none'](),
       },
diff --git a/packages/frontend/core/src/desktop/dialogs/setting/general-setting/editor/edgeless/shape.tsx b/packages/frontend/core/src/desktop/dialogs/setting/general-setting/editor/edgeless/shape.tsx
index 283fc7f5b5..84d6bcc3d7 100644
--- a/packages/frontend/core/src/desktop/dialogs/setting/general-setting/editor/edgeless/shape.tsx
+++ b/packages/frontend/core/src/desktop/dialogs/setting/general-setting/editor/edgeless/shape.tsx
@@ -107,16 +107,24 @@ export const ShapeSettings = () => {
     () => [
       {
         value: StrokeStyle.Solid,
+        testId: 'shape-border-style-solid-trigger',
         label:
           t['com.affine.settings.editorSettings.edgeless.note.border.solid'](),
       },
       {
         value: StrokeStyle.Dash,
+        testId: 'shape-border-style-dash-trigger',
         label:
           t['com.affine.settings.editorSettings.edgeless.note.border.dash'](),
       },
+      {
+        value: StrokeStyle.Dot,
+        testId: 'shape-border-style-dot-trigger',
+        label: 'Dot',
+      },
       {
         value: StrokeStyle.None,
+        testId: 'shape-border-style-none-trigger',
         label:
           t['com.affine.settings.editorSettings.edgeless.note.border.none'](),
       },
